### PR TITLE
KUBE-380: Clean up volume attachments

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

File: actions/delete_node_handler.go [hunk 1]
```diff
@@ -156,10 +159,40 @@ func (h *deleteNodeHandler) Handle(ctx context.Context, action *castai.ClusterAc
 			if len(pods.Items) > 0 {
 				return true, fmt.Errorf("waiting for %d pods to be terminated on node %v", len(pods.Items), req.NodeName)
 			}
+			volumeAttachments, err := h.clientset.StorageV1().VolumeAttachments().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return true, err
+			}
+			if err != nil {
+				return true, fmt.Errorf("unable to list volume attachments for node %q err: %w", req.NodeName, err)
+			}
```
Comment (Bruno-DaSilva) [inline actions/delete_node_handler.go:168]: Redundant block, only the first one will ever trigger (both err != nil)

**Review by Bruno-DaSilva — CHANGES_REQUESTED** [review]

**Comment by Bruno-DaSilva** [comment]
thanks for the quick fix!

File: actions/delete_node_handler.go (hunk 1 above)
Comment (stgleb) [reply actions/delete_node_handler.go]: good catch, thanks

**Review by Bruno-DaSilva — APPROVED** [review]

File: actions/delete_node_handler.go [hunk 2]
```diff
@@ -156,10 +159,37 @@ func (h *deleteNodeHandler) Handle(ctx context.Context, action *castai.ClusterAc
 			if len(pods.Items) > 0 {
 				return true, fmt.Errorf("waiting for %d pods to be terminated on node %v", len(pods.Items), req.NodeName)
 			}
+			volumeAttachments, err := h.clientset.StorageV1().VolumeAttachments().List(ctx, metav1.ListOptions{})
```
Comment (aldor007) [inline actions/delete_node_handler.go:162]: Do we need to have more permissions for this one?

Comment (aldor007) [reply actions/delete_node_handler.go:162]:
```
time="2024-05-27T13:54:55Z" level=error msg="handle actions" func="github.com/castai/cluster-controller/actions.(*service).handleActions.func1()" file="actions.go:190" ctrl_pod_name=castai-cluster-controller-8b4f5c4bf-xgnd6 error="handling action *castai.ActionDeleteNode: deleting volume attachments: volumeattachments.storage.k8s.io is forbidden: User \"system:serviceaccount:castai-agent:castai-cluster-controller\" cannot list resource \"volumeattachments\" in API group \"storage.k8s.io\" at the cluster scope" id=bdf3c818-17b1-4bc0-8074-bf3d07b08135 k8s_version=1.27+ running_on=ip-10-0-2-23.eu-central-1.compute.internal version=commit-7e7399f5724403539da0f4c4563175aab4df73be
````

Comment (stgleb) [reply actions/delete_node_handler.go:163]: added additional checks for forbidden error

**Review by andrejatcastai — APPROVED** [review]

**Pull request merged**